src/Ng/CommandHelpers.cs

```diff
@@ -45,86 +114,113 @@ private static void TraceRequiredArgument(string name)
             Trace.TraceError("Required argument \"{0}\" not provided", name);
         }
 
-        public static string GetSource(IDictionary<string, string> arguments)
+        private static void TraceMissingArgument(string name)
         {
-            string value;
-            if (!arguments.TryGetValue("-source", out value))
+            Console.WriteLine("Argument \"{0}\" not provided", name);
+            Trace.TraceWarning("Argument \"{0}\" not provided", name);
+        }
+
+        private static void TryGetArgument(IDictionary<string, string> arguments, string searchArg, out string value, bool required = false)
+        {
+            if (!arguments.TryGetValue("-" + searchArg, out value))
+            {
+                if (required)
+                {
+                    TraceRequiredArgument(searchArg);
+                    throw new ArgumentException();
+                }
+                else
+                {
+                    TraceMissingArgument(searchArg);
+                }
+
+                value = null;
+            }
+        }
+
+        public static SecretInjector GetSecretInjector(IDictionary<string, string> arguments)
```

Comment:
>GetSecretInjector [](start = 37, length = 17)

In this method you don't distinguish between two cases:
1. We don't want to use KeyVault, so we didn't provide any arguments for it
2. We WANT to use KeyVault, but we provided partial arguments for it.
In the first case, the method should succeed, and the flow should continue
In the second case. we want to stop the flow, because we got bad arguments